drivers/volume/kdmp/kdmp.go

```diff
@@ -749,12 +751,22 @@ func (k *kdmp) StartRestore(
 			pvc.Name = bkpvInfo.PersistentVolumeClaim
 			pvc.Namespace = restoreNamespace
 		}
-		// Check if the snapshot size is larger than the pvc size.
-		// Update the pvc size to that of the snapshot size if so.
+		// Below two cases, we will change the PVC original size.
+		// case1: If the snapshot size is greater than the PVC size. In some of the filesystem
+		// provisioned PVC, the content of the PVC is allowed to grow beyond the PVC size.
+		// case2: If the snapshot size + 10% of the snapshot size goes beyind the PVC size.
+		// We also allow the customer to configure the sizePercentage to configured, if the 10% does not
+		// satisfy their storage provisioner.
+		// KDMP_RESTORE_PVC_SIZE_PERCENTAGE parameter need to be added to the kdmp-config configmap in kube-system namespace.
+		sizePercentage := utils.GetRestorePvcSizePercetage()
+		if sizePercentage == 0 {
+			sizePercentage = defaultPvcSizeIncreasePercentage
+		}
+		logrus.Infof("StartRestore: sizePercentage is - %v", sizePercentage)
+		newPvcQuantity := resource.NewQuantity(int64(bkpvInfo.TotalSize)+((int64(sizePercentage)*int64(bkpvInfo.TotalSize))/int64(100)), resource.BinarySI)
 		pvcQuantity := pvc.Spec.Resources.Requests[v1.ResourceStorage]
-		if pvcQuantity.CmpInt64(int64(bkpvInfo.TotalSize)) == -1 {
-			newPvcQuantity := resource.NewQuantity(int64(bkpvInfo.TotalSize), resource.BinarySI)
-			logrus.Debugf("setting size of pvc %s/%s same as snapshot size %s", pvc.Namespace, pvc.Name, newPvcQuantity.String())
+		if (pvcQuantity.CmpInt64(int64(bkpvInfo.TotalSize)) == -1) || (newPvcQuantity.CmpInt64(int64(bkpvInfo.TotalSize)) == 1) {
```

Comment:
(newPvcQuantity.CmpInt64(int64(bkpvInfo.TotalSize)) == 1) this condition will always be true cos newPvcQuantity is with bkpvInfo.TotalSize+sizePercent since its || we will update the size always here. This I believe will reduce the PVC to lower than the PVC set by default. 

I think we should be checking for 
```
if (pvcQuantity.CmpInt64(int64(newPvcQuantity.TotalSize)) == -1 {
    logrus.Debugf("StartRestore: setting size of pvc %s/%s  to accommodate snapshot size with buffer %s", pvc.Namespace, pvc.Name, newPvcQuantity.String())
    pvc.Spec.Resources.Requests[v1.ResourceStorage] = *newPvcQuantity
 }
 ```

Reply:
Got it. Changed the logic. Thanks